### ISSUE TITLE
[JDK8 OJDK-MH] getCallerClass failures

### ISSUE THREAD

**Issue by babsingh** [eclipse-openj9/openj9#14989]
The below failure(s) is(are) only seen when OJDK MHs are enabled in OpenJ9.

### Errors

```
Testing: call APIs with/without @CallerSensitive annotation from the bootstrap classloader
Test start time: 2022/04/20 01:26:20 Eastern Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_ojdk292_Personal_testList_0/openjdkbinary/j2sdk-image/bin/java"   -Xdump -Xbootclasspath/a:/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_ojdk292_Personal_testList_0/aqa-tests/TKG/../../jvmtest/functional/cmdline_options_testresources/cmdlinetestresources.jar com.ibm.j9.getcallerclass.TestGroup bootClassLoader
Time spent starting: 3 milliseconds
Time spent executing: 236 milliseconds
Test result: FAILED
Output from test:
 [OUT] test_getCallerClass_fromBootExtWithAnnotation: PASSED
 [OUT] test_ensureCalledFromBootstrapClass_fromBootWithAnnotation: PASSED
 [OUT] test_getCallerClass_Helper_Reflection_fromBootExtWithAnnotation: PASSED: return com.ibm.j9.getcallerclass.RefectionMHTests
 [OUT] test_getCallerClass_Direct_Reflection_fromBootExtClassLoader: PASSED
 [OUT] test_getCallerClass_Helper_MethodHandle_fromBootExtWithAnnotation: FAILED: return java.lang.invoke.MethodHandleImpl$BindCaller$T/0x0000000000000000
 [OUT] test_getCallerClass_Direct_MethodHandle_fromBootExtClassLoader: PASSED
 [OUT] test_getCallerClass_MethodHandle_ArgumentHelper: FAILED: return java.lang.invoke.MethodHandleImpl$BindCaller$T/0x0000000000000000
 [OUT] test_getCallerClass_fromBootExtWithoutAnnotation: PASSED
 [OUT] test_ensureCalledFromBootstrapClass_fromBootExtWithoutAnnotation: PASSED
>> Success condition was not found: [Output match: All Tests Completed and Passed]
>> Failure condition was found: [Output match: FAILED]
>> Failure condition was not found: [Output match: Unhandled Exception]
>> Failure condition was not found: [Output match: corrupt]
>> Failure condition was not found: [Output match: Processing dump event]
>> Failure condition was not found: [Output match: JVM requested Java dump]
>> Failure condition was not found: [Output match: JVM requested Snap dump]

Testing: call APIs with/without @CallerSensitive annotation from the extension classloader
Test start time: 2022/04/20 01:26:20 Eastern Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_ojdk292_Personal_testList_0/openjdkbinary/j2sdk-image/bin/java"   -Xdump -Djava.ext.dirs=/home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_ojdk292_Personal_testList_0/aqa-tests/TKG/../../jvmtest/functional/cmdline_options_testresources com.ibm.j9.getcallerclass.TestGroup extClassLoader
Time spent starting: 3 milliseconds
Time spent executing: 184 milliseconds
Test result: FAILED
Output from test:
 [OUT] test_getCallerClass_fromBootExtWithAnnotation: PASSED
 [OUT] test_ensureCalledFromBootstrapClass_fromExtWithAnnotation: PASSED
 [OUT] test_getCallerClass_Helper_Reflection_fromBootExtWithAnnotation: PASSED: return com.ibm.j9.getcallerclass.RefectionMHTests
 [OUT] test_getCallerClass_Direct_Reflection_fromBootExtClassLoader: PASSED
 [OUT] test_getCallerClass_Helper_MethodHandle_fromBootExtWithAnnotation: FAILED: return java.lang.invoke.MethodHandleImpl$BindCaller$T/0x0000000000000000
 [OUT] test_getCallerClass_Direct_MethodHandle_fromBootExtClassLoader: PASSED
 [OUT] test_getCallerClass_MethodHandle_ArgumentHelper: FAILED: return java.lang.invoke.MethodHandleImpl$BindCaller$T/0x0000000000000000
 [OUT] test_getCallerClass_fromBootExtWithoutAnnotation: PASSED
 [OUT] test_ensureCalledFromBootstrapClass_fromBootExtWithoutAnnotation: PASSED
>> Success condition was not found: [Output match: All Tests Completed and Passed]
>> Failure condition was found: [Output match: FAILED]
>> Failure condition was not found: [Output match: Unhandled Exception]
>> Failure condition was not found: [Output match: corrupt]
>> Failure condition was not found: [Output match: Processing dump event]
>> Failure condition was not found: [Output match: JVM requested Java dump]
>> Failure condition was not found: [Output match: JVM requested Snap dump]
```

### Failing Test Targets

- `BUILD_LIST=functional`
    - `cmdLineTester_getCallerClassTests_SE80_0`

### How-to Run Tests?

https://github.com/eclipse/openj9/blob/master/test/docs/OpenJ9TestUserGuide.md

### Steps to build an OpenJ9 JDK8 with OJDK MHs enabled

Refer to https://github.com/eclipse-openj9/openj9/issues/14541.

**Comment by babsingh** [comment]
**TODO:** Check RI behaviour for the failing test. The test may not be checking the correct behaviour.

**Comment by ThanHenderson** [comment]
This also affects JDK11. https://github.com/eclipse-openj9/openj9/pull/18372 supports JDK11 but more investigation is needed for JDK8.

**Comment by ThanHenderson** [comment]
Closed by #18396
